### PR TITLE
Subject questionnaire

### PR DESCRIPTION
**Pull request by Aharonyn**

### CODE REVIEW

File: pylabber/settings.py [hunk 1]
```diff
@@ -32,6 +32,9 @@
     DB_HOST=(str, "localhost"),
     DB_PORT=(int, 5432),
     RAW_SUBJECT_TABLE_PATH=(str, "subjects.xlsx"),
+    SUBJECT_IDS_TABLE=(str, "subjects_ID.xlsx"),
```
Comment (ZviBaratz) [inline pylabber/settings.py]: No need for both.

File: pylabber/utils/utils.py [hunk 2]
```diff
@@ -72,3 +74,28 @@ def get_prep_value(self, value):
...
+        io.BytesIO
+            Decrypted excel.
+        """
+
+        f = msoffcrypto.OfficeFile(open(filename, "rb"))
+        f.load_key(password=password)
+        decrypted = io.BytesIO()
+        f.decrypt(decrypted)
+
+        return decrypted
```
Comment (ZviBaratz) [inline pylabber/utils/utils.py]: Wouldn't this be better placed within the `QuestionnaireReader` class? I can't see why this additional class is required.

File: research/models/subject.py [hunk 3]
```diff
@@ -143,3 +146,57 @@ def get_raw_information(self) -> pd.Series:
...
+            right_on="ID",
+        )
+
+        # Merging tables to get the questionnaire data.
+        output = output.merge(
+            questionnaire,
+            how="left",
+            left_on="Questionnaire",
+            right_on=questionnaire.index,
+        )
```
Comment (ZviBaratz) [inline research/models/subject.py]: This doesn't belong here. There should be an external function to create the merged dataframe, and this method should simply call it and return the appropriate index.

**Review by ZviBaratz — CHANGES_REQUESTED** [review]
Requires some reformatting. I think the merged dataframe should either be created in a separate function or externally to the application altogether, so that the _subjects.xlsx_ file already contain the required ID and associated info.

File: pylabber/settings.py (hunk 1 above)
Comment (Aharonyn) [reply pylabber/settings.py]: Corrected.

File: pylabber/utils/utils.py (hunk 2 above)
Comment (Aharonyn) [reply pylabber/utils/utils.py]: Removed this method.

File: research/models/subject.py (hunk 3 above)
Comment (Aharonyn) [reply research/models/subject.py]: Moved to an external function.

**Review by ZviBaratz — APPROVED** [review]
LGTM :+1:

**Pull request merged**